From 3491e31db25228f2b3bfb4e9c7702a466f8d98e9 Mon Sep 17 00:00:00 2001
From: Marcos Bento <Marcos.Bento@ecmwf.int>
Date: Thu, 1 Jun 2023 11:01:41 +0200
Subject: [PATCH] Remove unused Timer class

n.b. this allows compiling in old MacOSX platforms (e.g. 10.9 @ conda-forge) which don't defined CLOCK_REALTIME
---
 Http/src/ApiV1Impl.cpp | 21 ---------------------
 1 file changed, 21 deletions(-)

diff --git a/Http/src/ApiV1Impl.cpp b/Http/src/ApiV1Impl.cpp
index 67511eeec..28ece19cd 100644
--- a/Http/src/ApiV1Impl.cpp
+++ b/Http/src/ApiV1Impl.cpp
@@ -19,7 +19,6 @@
 #include <mutex>
 #include <string>
 #include <thread>
-#include <time.h>
 
 #include <condition_variable>
 
@@ -57,24 +56,6 @@ const char* ECF_PASS = getenv("ECF_PASS");
 
 namespace {
 
-class Timer {
-public:
-    Timer() : Timer("") {}
-    Timer(const std::string& name) : name_(name) { clock_gettime(CLOCK_REALTIME, &start_ts); }
-    ~Timer() {
-        clock_gettime(CLOCK_REALTIME, &stop_ts);
-        int64_t start = start_ts.tv_sec * 1000000000 + start_ts.tv_nsec;
-        int64_t stop  = stop_ts.tv_sec * 1000000000 + stop_ts.tv_nsec;
-
-        printf("Timer %s %ld ms\n", name_.c_str(), (stop - start) / 1000 / 1000);
-    }
-
-private:
-    std::string name_;
-    timespec start_ts;
-    timespec stop_ts;
-};
-
 template <typename T>
 void trigger_defs_update(T&& func) {
     {
@@ -581,8 +562,6 @@ json update_node_definition(const httplib::Request& request) {
     // user can give us just the definition of a new task, and
     // we will add that to the suite.
 
-    // Timer t(path);
-
     // this takes a long time, but we need a copy as we are changing
     // the defs locally here
 
-- 
2.40.1

